hpcbench/benchmark/minigemm.py

```diff
@@ -0,0 +1,261 @@
+"""Test basic functionality of BB5
+"""
+from __future__ import print_function
+
+import os
+import stat
+import subprocess
+
+from cached_property import cached_property
+
+from hpcbench.api import (
+    Benchmark,
+    Metric,
+    Metrics,
+    MetricsExtractor)
+
+
+class GemmExtractor(MetricsExtractor):
+
+    METRICS = dict(
+        time=Metrics.Second,
+        gflops=Metrics.GFlops,
+        checksum=Metric('#', float)
+    )
+
+    @property
+    def metrics(self):
+        """ The metrics to be extracted.
+            This property can not be replaced, but can be mutated as required
+        """
+        return self.METRICS
+
+    def extract_metrics(self, metas):
+        # parse stdout and extract desired metrics
+        with open(self.stdout) as istr:
+            timel = istr.readline()
+            time = float(timel.split()[-1][:-1])
+            flopl = istr.readline()
+            gflops = float(flopl.split()[0])
+            checkl = istr.readline()
+            chk = float(checkl.split()[-1])
+        metrics = dict(
+            time=time,
+            gflops=gflops,
+            checksum=chk
+        )
+        return metrics
+
+
+SOURCE = """#include <iostream>
```

Comment:
```
SOURCE = """\
#include <iostream>
```

You can also use `textwap.dedent` if you want to have c++ code indented. (there is an example in hpcbench).